### PR TITLE
ruby: initial verification lib

### LINKED ISSUE #2: Create Initial Ruby Library
Create initial ruby project structure & webhook verification functionality

### PR DESCRIPTION
**Pull request by svix-frank**
I'm not very familiar with ruby myself but I've added rspec tests which should cover all our bases.

To test you can run:
```sh
bundle install
bundle exec rspec spec
```
We will probably want to add this to `rubygems.org`

Example Usage:
```ruby
require 'svix'

wh = Svix::Webhook.new("MfKQ9r8GKYqrTwjUPD8ILPZIo2LaLaSw")
payload = '{"test": 2432232314}'
headers = {
  "svix-id" => "msg_p5jXN8AQM9LWM0D4loKWxJek",
  "svix-timestamp" => "1614265330",
  "svix-signature" => "v1,g0hM9SsE+OTPJTGt/tmIKtSyZlE3uFJELVlNIOLJ1OE="
}
json = wh.verify(payload, headers)
```

Closes #2.

### CODE REVIEW

**Review by tasn — APPROVED** [review]
I'm not qualified to review ruby, but it looks alright!

File: ruby/lib/svix/webhook.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            toSign = "#{msgId}.#{msgTimestamp}.#{payload}"
+            signature = Base64.encode64(OpenSSL::HMAC.digest(OpenSSL::Digest.new('sha256'), @secret, toSign)).strip()
+
+            passedSignatures = msgSignature.split(" ")
+            passedSignatures.each do |versionedSignature|
+                version, expectedSignature = versionedSignature.split(',', 2)
+                if version != "v1"
+                    next
+                end
+                if signature == expectedSignature
```
Comment (tasn) [inline ruby/lib/svix/webhook.rb:26]: After reviewing the php version I realised I missed it here too! Needs to use a secure comparison function.

**Review by tasn — CHANGES_REQUESTED** [review]

File: ruby/lib/svix/webhook.rb (hunk 1 above)
Comment (svix-frank) [reply ruby/lib/svix/webhook.rb]: Borrowed the secure cmp from Rails ActiveSupport

**Review by tasn — APPROVED** [review]
Looks good to me.

**Comment by svix-frank** [comment]
We should be good to merge this https://rubygems.org/gems/svix 🚀

**Pull request merged**